src/gluonnlp/data/dataset.py

```diff
@@ -296,3 +297,38 @@ def get_field(self, field):
         """
         idx = self._keys.index(field)
         return self._data[idx]
+
+
+class JsonlDataset(SimpleDataset):
```

Comment:
This is not a concrete class. let's prefix it with underscore `_` and use internally only